### ISSUE TITLE
Gradient of covariance propagation of IMU integrator is not trackable.

### ISSUE THREAD

**Issue by wang-chen** [pypose/pypose#37]
### Summary

- The current implementation is based on creating new tensors without `requires_grad=True`, see [here](https://github.com/pypose/pypose/blob/53ea256fde8ba9358e955d0b52e8fdee0c4c3198/pypose/module/imu_preintegrator.py#L111). However, if set `requires_grad=True`, then directly inplace setting its value is not permitted, like [here](https://github.com/pypose/pypose/blob/53ea256fde8ba9358e955d0b52e8fdee0c4c3198/pypose/module/imu_preintegrator.py#L114).
- Passing covariance as an argument is not working, see [here](https://github.com/pypose/pypose/blob/53ea256fde8ba9358e955d0b52e8fdee0c4c3198/pypose/module/imu_preintegrator.py#L99). Interfaces and codes need to be rephrased.

### Possible fixes

- Consider to use `torch.stack` or `torch.cat` to retain its gradients, like [here](https://github.com/pypose/pypose/blob/53ea256fde8ba9358e955d0b52e8fdee0c4c3198/pypose/lietensor/basics.py#L37).

- Assume `x = torch.zeros()`, use `x[i]= x[i] +  y`, instead of `x[i]=  y` for value setting.
